cmd/bundle/main.go

```diff
@@ -0,0 +1,124 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/shipwright-io/build/pkg/reconciler/buildrun/resources/sources"
+	"github.com/spf13/pflag"
+)
+
+var flagValues struct {
+	image      string
+	target     string
+	secretPath string
+}
+
+func init() {
+	pflag.StringVar(&flagValues.image, "image", "", "Location of the bundle image (mandatory)")
+	pflag.StringVar(&flagValues.target, "target", "/workspace/source", "The target directory to place the code")
+	pflag.StringVar(&flagValues.secretPath, "secret-path", "", "A directory that contains access credentials (optional)")
+}
+
+func main() {
+	if err := Do(context.Background()); err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+// Do is the main entry point of the bundle command
+func Do(ctx context.Context) error {
+	pflag.Parse()
+
+	if flagValues.image == "" {
+		return fmt.Errorf("mandatory flag --image is not set")
+	}
+
+	ref, err := name.ParseReference(flagValues.image)
+	if err != nil {
+		return err
+	}
+
+	auth, err := resolveAuthBasedOnTarget(ref)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("Pulling image %q", ref)
+	if err := sources.Unbundle(
+		ref,
+		flagValues.target,
+		remote.WithContext(ctx),
+		remote.WithAuth(auth)); err != nil {
+		return err
+	}
+
+	log.Printf("Image content was extracted to %s\n", flagValues.target)
+	return nil
+}
+
+func resolveAuthBasedOnTarget(ref name.Reference) (authn.Authenticator, error) {
+	// In case no secret is mounted, use anonymous
+	if flagValues.secretPath == "" {
+		log.Printf("No access credentials provided, using anonymous mode")
+		return authn.Anonymous, nil
+	}
+
+	// Read the registry credentials from the well-known location
+	file, err := os.Open(filepath.Join(flagValues.secretPath, ".dockerconfigjson"))
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	cf, err := config.LoadFromReader(file)
+	if err != nil {
+		return nil, err
+	}
+
+	// Look-up the respective registry server inside the credentials
+	var registryName = ref.Context().RegistryStr()
```

Comment:
If you intentionally used `var` here, I'd be interested in the pattern.

```suggestion
	registryName := ref.Context().RegistryStr()
```